Library/Homebrew/formula.rb

```diff
@@ -957,8 +957,14 @@ def unlock
   end
 
   def migration_needed?
-    oldname && !rack.exist? && (dir = HOMEBREW_CELLAR/oldname).directory? &&
-      !dir.subdirs.empty? && tap == Tab.for_keg(dir.subdirs.first).tap
+    return false unless oldname
+    return false if rack.exist?
+
+    old_cellar_dir = HOMEBREW_CELLAR/oldname
+
+    return false unless old_cellar_dir.directory?
+    return false if old_cellar_dir.subdirs.empty?
```

Comment:
That's a tiny detail, but I think since this and the line above reference `old_cellar_dir` in their condition, it would be more logical to group them with the `old_cellar_dir =` line than with the line below, i.e. that's how I would expect the empty lines to be distributed:

```ruby
return false unless oldname
return false if rack.exist?

old_cellar_dir = HOMEBREW_CELLAR/oldname
return false unless old_cellar_dir.directory?
return false if old_cellar_dir.subdirs.empty?

tap == Tab.for_keg(old_cellar_dir.subdirs.first).tap
```

Does that make sense?